Documentation/writing-principles-and-guidelines/spelling-and-typo3-terminology.rst

```diff
@@ -1,111 +1,1006 @@
+        functionalities and tools. Modules usually come in the form of
+        "Backend Modules", and are part of the left-hand side navigation.
+
+    ..  _multisite:
+    -   *Multisite*
+
+        Multisite refers to the capability of TYPO3 to manage multiple distinct
+        websites from a single installation.
+
+*   O
+
+    ..  _overrides:
+    -   *Overrides*
+
+        Overrides, specifically "TCA Overrides", allow TYPO3 extensions to
+        change central configuration of records and content elements.
+
+*   P
+
+    ..  _package:
+    -   *Package*
+
+        A Package is a bundle of files and resources used for installing and
+        configuring extensions or functionalities in TYPO3. Commonly, TYPO3
+        extensions are provided as "Composer Packages", where this term is the
+        short form of.
+
+    ..  _page-frame---tree-frame---module-frame---navigation-frame:
+    -   *Page Frame / Tree Frame / Module Frame / Navigation Frame*
+
+        Page frame, Tree frame, and Module frame are various sections in the
+        TYPO3 backend used for displaying and navigating content and modules.
+
+    ..  _page-tree:
+    -   *Page Tree*
+
+        The Page Tree is a hierarchical representation of the page structure in
+        the TYPO3 backend used for navigating and managing the website. It is
+        shown in the middle section of the TYPO3 Backend in the places where
+        content is edited.
+
+    ..  _page-view:
+    -   *Page View*
+
+        The Page View is a view in the TYPO3 backend used for editing and
+        managing individual page content.
+
+    ..  _pagebuilder:
+    -   *Pagebuilder*
+
+        Pagebuilders are tools in TYPO3 for creating and designing page layouts
+        and content. Pagebuilders are often used to create "Sitepackage
+        extensions", which define the central TYPO3 frontend look and the
+        definitions of content elements. Since these sitepackages can often be
+        repetitive and contain boilerplate code, Pagebuilders can help to
+        auto-generate these sitepackages.
+
+    ..  _pagerenderer:
+    -   *PageRenderer*
+
+        The PageRenderer is a PHP API component in TYPO3 responsible for
+        rendering and displaying page content in the frontend.
+
+    ..  _palette--tca:
+    -   *Palette (TCA)*
+
+        A Palette in the TCA (Table Configuration Array) is a grouping of fields
+        that can be displayed and edited together.
+
+    ..  _partial:
+    -   *Partial*
+
+        A Partial is a re-usable component of Fluid templates, that can be
+        parametrized.
+
+    ..  _permissions---acl:
+    -   *Permissions / ACL*
+
+        Permissions and Access Control Lists (ACL) are mechanisms in TYPO3 for
+        managing access rights and restrictions for users and groups.
+
+    ..  _pibase:
+    -   *piBase*
+
+        piBase is a base class for developing frontend plugins in TYPO3.
```

Comment:
The name "piBase" is based on the old class `class.tslib_pibase.php`. With the upgrade to TYPO3 6.0 various classes were renamed and `class.tslib_pibase.php` is now called `AbstractPlugin.php`.
The AbstractPlugin brings a lot of functionality like translation, fetching data and pagination. This class has to be extended by your own implementation and ideas of your plugin.